lib/AckGroupingTracker.cc

```diff
@@ -84,29 +94,43 @@ void AckGroupingTracker::doImmediateAck(const std::set<MessageId>& msgIds, Resul
         return;
     }
 
+    std::set<MessageId> ackMsgIds;
+
+    for (const auto& msgId : msgIds) {
+        if (auto chunkMessageId =
+                std::dynamic_pointer_cast<ChunkMessageIdImpl>(Commands::getMessageIdImpl(msgId))) {
+            auto msgIdList = chunkMessageId->moveChunkedMessageIds();
+            doImmediateAck(std::set<MessageId>(std::make_move_iterator(msgIdList.begin()),
+                                               std::make_move_iterator(msgIdList.end())),
+                           callback);
```

Comment:
We should add `ChunkMessageIdImpl::chunkedMessageIds_` into the `ackMsgIds` instead of calling `doImmediateAck`.

Reply:
+1 that will make the code more clear.